internal/controller/accountiam_controller.go

```diff
@@ -287,26 +334,67 @@ func (r *AccountIAMReconciler) reconcileOperandResources(ctx context.Context, in
 		}
 	}
 
+	// Temporary update issuer in platform-auth-idp configmap
+	klog.Infof("Updating platform-auth-idp configmap")
+	idpconfig := &corev1.ConfigMap{}
+	if err := r.Get(ctx, client.ObjectKey{Name: "platform-auth-idp", Namespace: instance.Namespace}, idpconfig); err != nil {
+		klog.Errorf("Failed to get configmap platform-auth-idp in namespace %s", instance.Namespace)
+		return err
+	}
+	idpconfig.Data["OIDC_ISSUER_URL"] = decodedData.DefaultIDPValue
+	if err := r.Update(ctx, idpconfig); err != nil {
+		klog.Errorf("Failed to update ConfigMap platform-auth-idp in namespace %s: %v", instance.Namespace, err)
+		return err
+	}
+
+	// Delete the platform-auth-service and platform-identity-provider pod to restart it
+	if err := r.restartAndCheckPod(ctx, instance.Namespace, "platform-auth-service"); err != nil {
+		return err
+	}
+	klog.Infof(" platform-auth-service pod is ready")
+
+	if err := r.restartAndCheckPod(ctx, instance.Namespace, "platform-identity-provider"); err != nil {
+		return err
+	}
+	klog.Infof(" platform-identity-provider pod is ready")
+
+	klog.Infof("MCSP operand resources created successfully")
 	return nil
 }
 
 func (r *AccountIAMReconciler) configIM(ctx context.Context, instance *operatorv1alpha1.AccountIAM) error {
 
-	logger := log.FromContext(ctx)
-	logger.Info("Creating IM Config Job")
-	object := &unstructured.Unstructured{}
-	var buffer bytes.Buffer
+	host, err := r.getHost(ctx, "account-iam", instance.Namespace)
+	if err != nil {
+		return err
+	}
+	klog.Infof("account-iam route host: %s", host)
+
+	BootstrapData.AccountIAMURL = "https://" + host
+	klog.Infof("Creating IM Config Job")
 	decodedData, err := r.decodeData(BootstrapData)
 	if err != nil {
 		return err
 	}
+	if err := r.InjectData(ctx, instance, res.IMConfigYamls, decodedData); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *AccountIAMReconciler) InjectData(ctx context.Context, instance *operatorv1alpha1.AccountIAM, manifests []string, bootstrapData BootstrapSecret) error {
 
-	for _, v := range res.IMConfigYamls {
-		manifest := v
+	var buffer bytes.Buffer
+	object := &unstructured.Unstructured{}
```

Comment:
Do we need to put `object` into the for loop? So it will be initialized for each manifest?